### ISSUE TITLE
[Question]Merge features

### ISSUE THREAD

**Issue by Fr-Dae** [Spatchy/Infinite-Craft-Plus-Plus#29]
hello, you're probably the most accomplished project around infinit craft.

Could you talk to the people in charge of the other projects and merge them all together?
Why not add a menu option to activate or deactivate some.

- https://addons.mozilla.org/fr/firefox/addon/better-infinite-craft/ (QoL Search + shift for copy)
- https://addons.mozilla.org/fr/firefox/addon/infinite-craft-data-manager/ (import /export data)
- https://addons.mozilla.org/fr/firefox/addon/infinite-craft-memorizer/ (craft history)

**Comment by MasonStooksbury** [comment]
I'm down if other parties are

My extension is using ManifestV2 because I'm a noob and didn't realize there was a V3

So if we do this, I'll need to make some edits to mine to make sure it works seamlessly with the existing codebase

**Comment by Spatchy** [comment]
@MasonStooksbury I've been talking with Mikarific from the [Infinite Craft Helper](https://github.com/Mikarific/InfiniteCraftHelper) userscript and we have semi-agreed on a simple way of communicating between projects. I have first-drafted [some documentation](https://github.com/Spatchy/Infinite-Craft-Plus-Plus/blob/chore/meta-tag-documentation/COMPAT.md) that hopefully gets across how I envision this system working. Comments, ideas and pull requests to the doc are welcome.

**Comment by MasonStooksbury** [comment]
@Spatchy, sounds great! I'll dig more into this later.

If you think my extension would be a good one to add more functionality to IC++, I'd love to tidy mine up and add it in!

**Comment by Spatchy** [comment]
@MasonStooksbury If you wanted to merge code from your project into Infinite Craft++ that's a whole different discussion. I'd certainly be open to it, but I have no experience with cross-repo merges like that so I'm not sure how possible it would be to maintain the git history from your repo (which would obviously be ideal), especially if you wanted to continue maintaining Infinite Craft Ingredient Manager separately. Let me know if you know anything about that.

**Comment by MasonStooksbury** [comment]
Good point. Cross-repo merges do get weird. If this were going to be a thing, I'd probably just submit a PR to your repo with my code added in and then archive my repo. I'd then add a link bad to yours (since yours is more fleshed out and pretty lol)

I don't care much about the history and stuff for mine, so I'm fine just merging my stuff into yours.

Again, the biggest issue with mine is that it's using Manifest V2 and yours is on V3. There's some significant changes that I'd need to make to mine in order for it to work seamlessly with what you already have, but it is doable

**Comment by Spatchy** [comment]
Just skimming your code, I don't think there's too much that would need to change. We would probably want to reduce the scope of the permissions and find alternative ways to do some things without certain permissions if possible. I get the feeling each new permission adds **significant** time to the Chrome Web Store's review process which can already be frustratingly slow (24 hrs+) compared to Firefox's (5-10 mins).

**Comment by MasonStooksbury** [comment]
Yeah I probably could have worked the permissions a little better. The only big ones that I definitely need are the ability to read/write from local storage and trigger downloads. Everything else should be pretty simple if it needs reworking
